### PR TITLE
Simplify implementation of SmagorinskyLilly + some other closure doc fixes

### PR DESCRIPTION
**Pull request by glwagner**
I noticed it was a bit convoluted. I refactored the viscosity computation to hopefully be a bit more readable and understandable.

### CODE REVIEW

**Comment by navidcy** [comment]
Is the `SmagorinskyLilly` tested? If so I'm happy to approve this PR.

File: src/TurbulenceClosures/turbulence_closure_utils.jl [hunk 1]
```diff
@@ -38,3 +35,4 @@ end
     i, j, k = @index(Global, NTuple)
     @inbounds κₑ[i, j, k] = calc_nonlinear_κᶜᶜᶜ(i, j, k, grid, closure, tracer, tracer_index, U)
 end
+
```
Comment (tomchor) [inline src/TurbulenceClosures/turbulence_closure_utils.jl:38]:
```suggestion
```

**Comment by tomchor** [comment]
@glwagner nice. I think it's indeed clearer.

@navidcy I think the `SmagLilly` closure is tested in [time-stepping](https://github.com/CliMA/Oceananigans.jl/blob/main/test/test_time_stepping.jl), [closures](https://github.com/CliMA/Oceananigans.jl/blob/main/test/test_turbulence_closures.jl) and in [nonhydrostatic regression](https://github.com/CliMA/Oceananigans.jl/blob/main/test/test_nonhydrostatic_regression.jl#L63-L72). But I think the only instance where we're testing actual values (and not just compilation of trivial (i.e. zero) values) is in 

https://github.com/CliMA/Oceananigans.jl/blob/8b42137e4224b86fe1f3f1744bd5076aa8821085/test/test_nonhydrostatic_regression.jl#L63-L72

Which does seem to pass, so I'm approving this.

**Review by tomchor — APPROVED** [review]

File: src/TurbulenceClosures/turbulence_closure_utils.jl (hunk 1 above)
Comment (navidcy) [reply src/TurbulenceClosures/turbulence_closure_utils.jl:38]: Damn I missed this.

**Pull request merged**